ostd/src/arch/x86/cpu/mod.rs

```diff
@@ -385,96 +388,166 @@ cpu_context_impl_getter_setter!(
     [gsbase, set_gsbase]
 );
 
-/// The floating-point state of CPU.
+pub(super) static CPU_FEATURES: Once<FeatureInfo> = Once::new();
+pub(super) static XSTATE_MAX_FEATURES: Once<u64> = Once::new();
+pub(super) static XSAVE_SIZE: Once<usize> = Once::new();
+
+const MAX_FXSAVE_AREA_SIZE: usize = 512;
+
+#[allow(unused)]
+#[repr(packed)]
 #[derive(Clone, Copy, Debug)]
-#[repr(C)]
+struct FxSaveArea {
+    /// 512-byte-long 16-byte-aligned memory region.
+    buffer: [u8; MAX_FXSAVE_AREA_SIZE],
+}
+
+/// Mask features which are restored when returning to user space.
+///
+/// X87 | SSE | AVX | OPMASK | ZMM_HI256 | HI16_ZMM
+const XFEATURE_MASK_USER_RESTORE: u64 = 0b1110_0111;
+
+const MAX_XSAVE_AREA_SIZE: usize = 4096;
+
+#[allow(unused)]
+#[repr(packed)]
+#[derive(Clone, Copy, Debug)]
+struct XSaveArea {
+    legacy: FxSaveArea,
+    features: u64,
+    compaction: u64,
+    reserved: [u64; 6],
+    buffer: [u8; MAX_XSAVE_AREA_SIZE - MAX_FXSAVE_AREA_SIZE - 64],
+}
+
+/// The FPU state save area of user task.
+///
+/// This could be used either for the legacy SSE/MMX FPU state format (as saved
+/// by `FXSAVE` and restored by the `FXRSTOR` instructions), or the modern FPU state
+/// format (as saved by the `XSAVE`` and restored by the `XRSTOR` instructions).
```

Comment:
Asterinas is written for modern hardware. For example, we have no plan to support 32-bit x86. Here I suppose that all modern x86-64 CPUs support `XSAVE` and `XRSTOR`. If this is the case, then we should not support the legacy format. This helps simplify the implementation.

Reply:
The osdk-test CI failed because the `XSAVE_SIZE` is zero. Therefore, I believe we should retain the use of `fxsave`.